src/hotspot/share/runtime/globals.hpp

```diff
@@ -1362,6 +1362,21 @@ const int ObjectAlignmentInBytes = 8;
           "-XX:MallocLimit=2g:oom"                                          \
           "-XX:MallocLimit=compiler:200m:oom,code:100m")                    \
                                                                             \
+                                                                            \
+  product(ccstr, RssLimit, nullptr, DIAGNOSTIC,                             \
+          "Limit to Resident Set Size. The JVM will periodically "          \
+          "check if that limit had been reached and, if true, "             \
+          "generate a fatal error. The limit can be either given as a "     \
+          "memory size (e.g. \"2G\") or as a percentage of "                \
+          "the total available memory on this machine or in this "          \
+          "container (e.g. \"-XX:RssLimit=80%%\"). A value of 0 (default) " \
+          "disables this feature.")                                         \
+                                                                            \
+  product(uint, RssLimitCheckInterval, 5000, DIAGNOSTIC,                    \
+          "If RssLimit is set, interval, in ms, at which the JVM will "     \
+          "check the process resident set size."                            \
+          range(10, UINT_MAX))                                              \
```

Comment:
Can we actually handle enrolling a periodic task with a UINT_MAX interval?

Reply:
No, we can't; I'll correct the limit.

Reply:
fixed